openid-4-verifiable-presentations-1_0.md

```diff
@@ -1464,6 +1464,180 @@ issuers in Self-Sovereign Identity ecosystems using TRAIN</title>
+        request:  {
+          "response_type": "vp_token",
+          "nonce": "n-0S6_WzA2Mj",
+          "client_metadata": {...},
+          "presentation_definition": {...}
+        }
+      }]
+    }
+  });
+} else {
+// fallback to other invocation mechanisms
+}
+```
+
+and this is a non-normative example of how the corresponding response:
+
+```JavaScipt
+const { data } = response;
+// data is a byte array that contains the JSON or JWE which needs to be parsed
+```
+
+There are a couple of benefits for OID4VP implementers (both Verifiers as well as Wallets) to adopt OID4VP with the Digital Credentials API. To start with, the API is a privacy-preserving alternative to the invocation of Wallets through URLs, especially custom schemes. The browser will ensure the invocation of a Wallet is only performed if confirmed by the user based on contextual information of the request and the sender. It also allows to select Wallets based on the credential types being requested and supported, respectively. As request and responses are sent and received through the API, the user will always return to the browser tab where she had started, which results in an improved user experience. And the security of OID4VP implenentations can also be enhanced signficantly. Cross-device requests benefit from the use of proximity checks through a combined use of BLE and a QR Code (similar to Passkeys). Furthermore, the Wallet is provided with information about the Verifier's URL as authenticated by the browser to the request as an additional signal, which can be used for phishing detection.  
+
+## Protocol
+
+The value of the `protocol` parameter of the W3C Digital Credentials API MUST be set to `urn:openid.net:oid4vp' for this profile.
+
+## Request
+
+The `request` parameter of the W3C Digital Credentials API MUST contain a valid OID4VP Authorization Request, where every Authorization Request parameter is represented as a top-level JSON member as shown in the following non-normative example:
+
+```json
+{
+  "client_id": "client.example.org",
+  ...
+  "response_type": "vp_token",
+  "nonce": "n-0S6_WzA2Mj",
+  "presentation_definition": {...}
+}
+```
+
+The following Authorization Request parameters are supported with this profile: 
+
+* `client_id`
+* `client_id_scheme`
+* `response_type`
+* `nonce`
+* `presentation_definition`
+* `client_metadata`
+* `request`
+
+The `client_id` and `client_id_scheme` MUST be omitted in unsigned requests. The wallet determines the Client Identifier from the origin as asserted by the Browser. 
+
+This profile introduces a new parameter `expected_origins`.
+
+* `expected_origins`: An array of strings, each of the strings representing an origin of the Verifier making the request. This parameter MUST only be used with signed requests. It relates the logical Client Identifier to the physical endpoints that are legit origins for requests on behalf of this Client Identifier and is used to detect request replay.
```

Comment:
What is the use case for more than one origin here?

Reply:
i think when discussed at IIW, the use-case was, when the entity owns/controls multiple domains, the verifier backend can generate the request that can be used by multiple domains (ex. amazon's backend generating a signed request that can be used by any amazon domain in the EU - amazon.de, amazon.fr, etc.), but now I am not really sure. i think @tplooker was arguing in favor of an array and might have more context..?

Reply:
we confirmed during the WG call that above explanation is correct. @danielfett does this make sense? can we resolve the issue?